test/int/cluster_test.go

```diff
@@ -392,6 +392,38 @@ var _ = Describe("AtlasCluster", func() {
 			})
 		})
 	})
+
+	Describe("Create cluster, user, delete cluster and check secrets are removed", func() {
+		createdCluster = mdbv1.DefaultGCPCluster(namespace.Name, createdProject.Name)
+		By(fmt.Sprintf("Creating the Cluster %s", kube.ObjectKeyFromObject(createdCluster)), func() {
+			Expect(k8sClient.Create(context.Background(), createdCluster)).ToNot(HaveOccurred())
+
+			Eventually(testutil.WaitFor(k8sClient, createdCluster, status.TrueCondition(status.ReadyType), validateClusterCreatingFunc()),
+				1800, interval).Should(BeTrue())
+
+			doCommonChecks()
+			checkAtlasState()
+		})
+
+		passwordSecret := buildPasswordSecret(UserPasswordSecret, DBUserPassword)
+		Expect(k8sClient.Create(context.Background(), &passwordSecret)).To(Succeed())
+
+		createdDBUser := mdbv1.DefaultDBUser(namespace.Name, "test-db-user", createdProject.Name).WithPasswordSecret(UserPasswordSecret)
+		By(fmt.Sprintf("Creating the Database User %s", kube.ObjectKeyFromObject(createdDBUser)), func() {
+			Expect(k8sClient.Create(context.Background(), createdDBUser)).ToNot(HaveOccurred())
+
+			Eventually(testutil.WaitFor(k8sClient, createdDBUser, status.TrueCondition(status.ReadyType)),
+				20, interval).Should(BeTrue())
```

Comment:
in my experience 20 seconds is often not enough, would recommend something around 80